datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -111,6 +111,8 @@ pub(crate) struct GroupedHashAggregateStream {
     /// first element in the array corresponds to normal accumulators
     /// second element in the array corresponds to row accumulators
     indices: [Vec<Range<usize>>; 2],
+    // buffer to be reused to store hashes
+    hashes_buffer: Vec<u64>,
```

Comment:
❤️  this is a good change -- thanks @Dandandan . Pretty soon there will be *no* allocations while processing each batch (aka the hot loop) 🥳  -- I think with https://github.com/apache/arrow-datafusion/pull/6888 we can get rid of the counts in the sum accumulator